code/modules/projectiles/firing.dm

```diff
@@ -40,6 +40,13 @@
 	if (!istype(targloc) || !istype(curloc) || !BB)
 		return 0
 	BB.ammo_casing = src
+
+	var/firing_dir
+	if(BB.firer)
+		firing_dir = BB.firer.dir
+	if(!BB.suppressed && firing_effect_type)
+		new firing_effect_type(get_turf(src), firing_dir)

```

Comment:
`PoolOrNew(firing_effect_type, list(get_turf(src), firing_dir))` though I still have no idea if it's performance efficient.